pkg/generator/caches.go

```diff
@@ -242,6 +242,20 @@ func generateListenersAndRouteConfigs(
 		if err != nil {
 			return nil, nil, err
 		}
+
+		// if a certificate is configured, add a new filter chain to TLS listener
+		if useHTTPSListenerWithOneCert() {
+			externalHTTPSEnvoyListenerWithOneCert, err := newExternalEnvoyListenerWithOneCert(
```

Comment:
Although `newExternalEnvoyListenerWithOneCert()` creates a listener, we don't use the Listener but just want to use the `FilterChain`. Should we introduce a new function (or split `NewHTTPSListener`) to create the Chain?

Reply:
Thanks for your suggestion :raised_hands:
You're right, I changed that in my last commit : https://github.com/knative-sandbox/net-kourier/pull/743/commits/81aadf13824a72321b9492e34a639d0ddfb7435b. It required some refacto, but nothing too difficult.